packages/common/src/utils/domUtils.ts

```diff
@@ -134,38 +140,43 @@ export function setControlValue(
   el: FormControl,
   value: FieldValue | FieldValue[]
 ): void {
-  if (!isInputElement(el)) return;
+  if (!isFormControl(el)) return;
   const fieldValue = value;
-  if (el.type === 'checkbox') {
-    const checkboxesDefaultData = fieldValue;
-    if (
-      typeof checkboxesDefaultData === 'undefined' ||
-      typeof checkboxesDefaultData === 'boolean'
-    ) {
-      el.checked = !!checkboxesDefaultData;
+
+  if (isInputElement(el)) {
```

Comment:
The diff is not so friendly here but at first sight I see you've just moved every check inside of an `if (isInputElement(el))` right? If so this looks perfect.

Reply:
Yep. You could add a `?w=1` query param to this diff's URL and it would remove the whitespace changes.